### PR TITLE
Use redux in react

### PR DESCRIPTION
**Pull request by cynthiainga**
Project requirement:

- [ ] First of all - make sure that you keep state immutable in reducers!
- [ ] Import Redux Provider and your store in the main component of your React app.
- [ ] Connect your app to the Redux store.
- [ ] Make sure that list of books displays books from Redux store.
-    Use useSelector to consume the state.
- [ ] In your React component responsible for adding new books, add functionality for "Add new" button click:
-    Import Redux useDispatch.
-    Set the data inputs in the local React state
-    Once your new book object is ready to be submitted to Redux store, dispatch a corresponding action (import Action Creator that you created before and use it here).
- [ ] In your React component responsible for removing books, add a functionality for "Remove" button click:
-   Import Redux useDispatch.
-   Dispatch a corresponding action (import Action Creator that you created before and use it here).
- [ ] In your React component responsible for categories, add a functionality for "Check status" button click:
- Import Redux useDispatch.
- Dispatch a corresponding action (import Action Creator that you created before and use it here).

### CODE REVIEW

**Comment by cynthiainga** [comment]
For the Add and Remove book functionality I added it before in the previous PR so here I add only the functionality of the check status button.

Thanks.

File: src/components/Form.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React, { useState } from 'react';
 import { useDispatch } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
 import { addBook } from '../redux/books/books';
 
 const Form = () => {
```
Comment (er-contreras) [inline src/components/Form.js:6]: - [ ] The form also should include an input for an Author to be displayed as you did with the title.

**Review by er-contreras — CHANGES_REQUESTED** [review]
Hi @cynthiainga, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

almost there

## Required Changes ♻️

- [ ] In your readme file it would be great if the MIT LICENSE can be below the title LICENSE
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Form.js (hunk 1 above)
Comment (cynthiainga) [reply src/components/Form.js:6]: Hello @eri8-9 thanks for the review, so for your suggestion to ass the input for an Author it's a good idea but in the origin design there is no input for the Author and they told us to reproduce the same design. I can add the input for the Author later.
Thanks :pray:

Comment (micheaol) [reply src/components/Form.js:6]: As requested by the previous code reviewer, please ensure that your New Book React Component has a local state with title, author according to the project requirement.

I believe this was a requirement in a previous milestone:
image

Therefore, I highly suggest you implement the author's input, as you will need it, and will make your life easier on the next milestones.

Explanation
The design that you can see in the sneak peek, is just for the styling part of the project.

Hints
You could shorten the "Book title" input, and in that space insert your "Author" input, next to it, and you will preserve your styling. Like this:
image

I hope this helps.

**Review by micheaol — CHANGES_REQUESTED** [review]
Hi @cynthiainga , 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by cynthiainga** [comment]
Hello @micheaol  thanks for the explanation, I already add it.

**Comment by cynthiainga** [comment]
And the required change for the previous reviewer was optional, it doesn't require another review, I think you could have approved my project and explained to me in more detail what the other reviewer meant.

**Comment by micheaol** [comment]
@cynthiainga ,

Thank you for implementing the changes requested, Kindly re-submit for review on your dashboard. The changes is not [OPTIONAL] it is part of the project requirement.

**Review by micheaol — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hi @cynthiainga , 

Great work on making the changes requested by the previous reviewer 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 
- You set up linter files and no errors. 💯 

well-1

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by cynthiainga** [comment]
Thanks for the review :pray:

**Comment by nidalaa** [comment]
>For the Add and Remove book functionality I added it before in the previous PR so here I add only the functionality of the check status button.

Hello @cynthiainga ,

a small request to you. When something like that happens, please provide a link to the code that you think about. It will make code reviewers life much easier :)

**Pull request merged**